tools/server-admin/src/intTest/java/org/projectnessie/tools/admin/cli/NessieServerAdminTestExtension.java

```diff
@@ -0,0 +1,175 @@
+import org.junit.jupiter.api.extension.BeforeEachCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.ExtensionContext.Namespace;
+import org.junit.jupiter.api.extension.ExtensionContext.Store;
+import org.junit.jupiter.api.extension.ExtensionContext.Store.CloseableResource;
+import org.junit.jupiter.api.extension.ParameterContext;
+import org.junit.jupiter.api.extension.ParameterResolutionException;
+import org.junit.jupiter.api.extension.ParameterResolver;
+import org.junit.platform.engine.ConfigurationParameters;
+import org.junit.platform.engine.UniqueId;
+import org.projectnessie.junit.engine.MultiEnvTestExtension;
+import org.projectnessie.quarkus.tests.profiles.BaseConfigProfile;
+import org.projectnessie.versioned.storage.common.config.StoreConfig.Adjustable;
+import org.projectnessie.versioned.storage.common.persist.Backend;
+import org.projectnessie.versioned.storage.common.persist.Persist;
+import org.projectnessie.versioned.storage.testextension.BackendTestFactory;
+
+/**
+ * A JUnit5 extension that sets up the execution environment for Nessie CLI tests.
+ *
+ * <p>A {@link Persist} instance is created and injected into tests for manipulating the test Nessie
+ * repository
+ *
+ * <p>The test Nessie repository is erased and re-created for each test case.
+ */
+public class NessieServerAdminTestExtension
+    implements BeforeAllCallback,
+        BeforeEachCallback,
+        AfterAllCallback,
+        ParameterResolver,
+        MultiEnvTestExtension {
+
+  private static final Namespace NAMESPACE = Namespace.create(NessieServerAdminTestExtension.class);
+  private static final String BACKEND = "backend";
+  private static final String BACKENDS_SYSTEM_PROPERTY = "backends";
+
+  @Override
+  public List<String> allEnvironmentIds(ConfigurationParameters configuration) {
+    String backendEnvs = System.getProperty(BACKENDS_SYSTEM_PROPERTY);
+    if (backendEnvs != null && !backendEnvs.isBlank()) {
+      return Arrays.asList(backendEnvs.split(","));
+    }
+    return Arrays.stream(NessieServerAdminTestBackends.values()).map(Enum::name).toList();
+  }
+
+  @Override
+  public String segmentType() {
+    return BACKEND;
+  }
+
+  static NessieServerAdminTestBackends environment(ExtensionContext context) {
+    return UniqueId.parse(context.getUniqueId()).getSegments().stream()
+        .filter(s -> BACKEND.equals(s.getType()))
+        .map(UniqueId.Segment::getValue)
+        .findFirst()
+        .map(NessieServerAdminTestBackends::valueOf)
+        .orElseThrow();
+  }
+
+  @Override
+  public void beforeAll(ExtensionContext context) throws Exception {
+    BackendHolder backend = getOrCreateBackend(context);
+
+    // Quarkus runtime will pick up relevant values from java system properties.
+    backend.config.forEach(System::setProperty);
+  }
+
+  @Override
+  public void afterAll(ExtensionContext context) throws Exception {
+    Quarkus.blockingExit();
+    System.gc();
```

Comment:
Why running GC?

Reply:
Whoops leftover, I will remove. I tried a few desperate things that didn't work (of course).